CRM/Pledge/BAO/PledgePayment.php

```diff
@@ -602,34 +615,44 @@ public static function calculateNextScheduledDate(&$params, $paymentNo, $basePay
    *   $statusId calculated status id of pledge
    */
   public static function calculatePledgeStatus($pledgeId) {
-    $paymentStatusTypes = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
+    $contributionStatuses = CRM_Core_PseudoConstant::get('CRM_Contribute_DAO_Contribution', 'contribution_status_id', array(
```

Comment:
Does the Contribution `buildOptions` function have a way to specify the column to fetch? By default, it returns the label, which will cause i18n problems.